Jetpack Pro Dashboard: Implement a popover to showcase the WPCOM hosting feature

yashwin opened the pull request:
Closes https://github.com/Automattic/jetpack-avalon/issues/48

## Proposed Changes

This PR implements a popover to showcase the WPCOM hosting feature on the Jetpack Pro Dashboard.

Also includes the following changes:

- Move the piece of code that deals with the rendering the buttons to a new component
- Minor code refactor

### Testing Instructions

**Prerequisites**

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

**Instructions**

1. Open the Jetpack Cloud live link.
2. Verify that the popover appears.

<img width="412" alt="Screenshot 2023-09-04 at 3 48 32 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/07fd127d-e919-49f7-a625-ecd6575f965b">


3. Click on the down arrow and verify that the popover is not shown and the dropdown items are shown.

<img width="405" alt="Screenshot 2023-09-01 at 2 13 06 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/3e6b3511-8fe5-43e0-a22b-d9628b5dddea">

4. Click the `Got it` button and verify that it is not shown again by refreshing the page
5. Verify that the buttons work as expected. 

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

yashwin committed:
Move the add licences and add site buttons to a new component

yashwin committed:
Minor code refactor

yashwin committed:
Create WPCOMHostingPopover component

yashwin committed:
Show WPCOM hosting feature popover on-hover

yashwin committed:
Update position and add the rocket icon

ilonagl commented:
This looks great, @yashwin! 

Could we consider displaying the popover as soon as a user accesses the dashboard, rather than triggering it only when hovering over the button? My concern is that users may not realize they can purchase WPCOM plans through this button. If the popover appears solely on hover, we risk some users overlooking this. WDYT?

Another note: I don't know how much effort it is, but is it possible to move the popover arrow closer to the button arrow (img below)? This would help users understand that they need to press the split button arrow button. WDYT?


Lastly, I fixed the line width to include WordPress.com in the first line. Sorry for overlooking this! 

<img width="335" alt="Screenshot 2023-09-04 at 12 56 06" src="https://github.com/Automattic/wp-calypso/assets/100145243/4688169d-5fcd-4483-a548-17cd943e7281">

yashwin committed:
Always show the popover and fix position

yashwin commented:
Thanks for your review, @ilonagl.

I have addressed all your comments. Could you please give it a try again?

<img width="372" alt="Screenshot 2023-09-04 at 3 49 31 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/ee3b3e0e-7ada-4b47-99cd-c3883960bf1e">

yashwin commented on client/components/split-button/index.jsx:
```diff
@@ -102,6 +103,8 @@ class SplitButton extends PureComponent {
 		const isEmptyOnClick = this.props.onClick === noop;
 		const onClick = isEmptyOnClick ? undefined : this.handleMainClick;
 
+		const popoverContextRef = popoverContext ?? this.popoverContext;
```
This is done so that we can place the tooltip to the dropdown instead to the main button

ilonagl reviewed (approved):
Fantastic work, @yashwin!

vitozev commented on client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-top-header-buttons/index.tsx:
```diff
@@ -0,0 +1,110 @@
+						onClick={ () =>
+							dispatch(
+								recordTracksEvent(
+									'calypso_jetpack_agency_dashboard_create_wpcom_atomic_site_button_click'
```
Do we need to track this event here? It looks like it has the same name as the event for the `Create a new WordPress.com site` button?

yashwin replied:
Good catch. I copied and pasted this part of the code from a different component but didn't catch it. 

I have made the changes here: https://github.com/Automattic/wp-calypso/pull/81311/commits/ef3b52342fbf1524c9adc07e0473283438aa948b

vitozev reviewed (approved):
LGTM!

yashwin committed:
Rename event names

yashwin committed:
Button should be primary

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8867496

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.
